controller/api/public/stat_summary.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/runconduit/conduit/pkg/k8s"
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	reqQuery = "sum(increase(response_total{%s}[%s])) by (%s, classification)"
+)
+
+type meshedCount struct {
+	inMesh    uint64
+	notInMesh uint64
+}
+
+func (h *handler) StatSummary(ctx context.Context, req *pb.StatSummaryRequest) (*pb.StatSummaryResponse, error) {
+	resourceType, err := k8s.CanonicalKubernetesNameFromFriendlyName(req.Selector.Resource.Type)
+	if err != nil {
+		return nil, err
+	}
+
+	switch resourceType {
+	case k8s.KubernetesDeployments:
+		return h.deploymentQuery(ctx, req)
+	default:
+		return nil, errors.New("Unimplemented resource type")
+	}
+}
+
+func (h *handler) deploymentQuery(ctx context.Context, req *pb.StatSummaryRequest) (*pb.StatSummaryResponse, error) {
+	rows := make([]*pb.StatTable_PodGroup_Row, 0)
+	deployments := []appsv1.Deployment{}
+	var meshCount map[string]*meshedCount
+
+	timeWindow, err := apiUtil.GetWindowString(req.TimeWindow)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: parallelize the k8s api query and the prometheus query
+	if req.Selector.Resource.Name == "" || req.Selector.Resource.Name == "all" {
+		deployments, meshCount, err = h.getDeployments(req.Selector.Resource.Namespace)
+	} else {
+		deployments, meshCount, err = h.getDeployment(req.Selector.Resource.Namespace, req.Selector.Resource.Name)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	requestLabels := buildRequestLabels(req)
+
+	requestMetrics, err := h.getRequests(ctx, requestLabels, req.Selector.Resource.Type, timeWindow)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, resource := range deployments {
+		row := pb.StatTable_PodGroup_Row{
+			Resource: &pb.Resource{
+				Namespace: resource.Namespace,
+				Type:      req.Selector.Resource.Type,
+				Name:      resource.Name,
+			},
+			TimeWindow: req.TimeWindow,
+			Stats:      requestMetrics[resource.Name],
+		}
+		if count, ok := meshCount[resource.Name]; ok {
+			row.MeshedPodCount = count.inMesh
+			row.TotalPodCount = count.inMesh + count.notInMesh
+		}
+
+		rows = append(rows, &row)
+	}
+
+	rsp := pb.StatSummaryResponse{
+		Response: &pb.StatSummaryResponse_Ok_{ // https://github.com/golang/protobuf/issues/205
+			Ok: &pb.StatSummaryResponse_Ok{
+				StatTables: []*pb.StatTable{
+					&pb.StatTable{
+						Table: &pb.StatTable_PodGroup_{
+							PodGroup: &pb.StatTable_PodGroup{
+								Rows: rows,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return &rsp, nil
+}
+
+func promLabel(key, val string) string {
+	return fmt.Sprintf("%s=\"%s\"", key, val)
+}
+
+func buildRequestLabels(req *pb.StatSummaryRequest) string {
+	labels := []string{}
+
+	var direction string
+	switch req.Outbound.(type) {
+	case *pb.StatSummaryRequest_None:
+		direction = "inbound"
```

Comment:
Since this case is also the default, I'd just omit the `StatSummaryRequest_None` case altogether.